nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -533,7 +535,18 @@ def load_checkpoint(self, checkpoint_path: Union[str, Path]) -> Dict[str, Any]:
                     self.lightning_module.get_model_module_list(), checkpoint_path, prefix="model."
                 )
 
-            sharded_state_dict = self.lightning_module.sharded_state_dict()
+            # [ModelOpt]: Initial loading from non-resume sharded checkpoint to a Distillation Model
```

Comment:
Since we only enable logit-distillation now, we don't have extra parameters right?. If so, let's remove this special handling here.